### PR TITLE
WIP/feedback request: add SSB tale:net

### PR DESCRIPTION
**Pull request by cryptix**
###### Motivation for this change

Opening this as a source-version of #46062 and to gather feedback on the approach. I'm still fairly novice in writing nix expressions and derivations but @NinjaTrappeur suggested I take over patchwork since I'm familiar with the project as a whole and have built an electron client myself. @samueldr also suggested in [this comment](https://github.com/NixOS/nixpkgs/pull/46062#issuecomment-419602936) that a source version would be preferable.

[tale:net](https://t4l3.net/concept/) is not http://github.com/ssbc/patchwork/ but very similar in architecture and spirit. They are both based on http://scuttlebutt.nz/ and are electron apps. Once I figure out the remaining questions, I can adapt the same recipe for patchwork and [patchbay](https://github.com/ssbc/patchbay/)

Also: This depends on #45997 but could be decoupled. I thought it would just be easier to test this with both new packages present.

ps: See #46000 for two standalone tools without electron

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Question:

* Is the _node2nix sub package_ approach sane?

Previously, I just used [this shell.nix](https://github.com/talenet/talenet/tree/master/shell.nix) to develop. In it, I can run `npm i && npm run build` (which is an alias for `node .electron-vue/build.js && electron-builder`) and `run dev` to make builts and development versions.

I discussed with @NinjaTrappeur that this invovles fetching a pre-built electron version from npmjs.com. I'd rather not do this.

In principle it is enough to run `node .electron-vue/build.js && cd ./dist/electron && electron .` and this can work on archlinux, when all the node and electron ABI versions correctly line up.

I'm having trouble understanding how to do the `npm i && node .electron-vue/build.js ` with node2nix, though. I generated the env from the source package.json and made some changes to it so that it doesn't import nixpkgs twice. The output looks fine but I'm not sure yet how to copy `node_modules` into this `out/` or if I should run the `node .electron-vue/build.js` in the embedded node env.

* re #45997: is `propagateBuildInputs` the correct method to have this package as a runtime-dependency to talenet or should add something to the fixupPhase of the node-env?

### CODE REVIEW

File: pkgs/applications/networking/ssb/talenet/default.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+, electron
+, nodejs-6_x
+# auxilary dep to scuttle-shell - needs to avail at runtime
+, systrayhelper
+}:
+
+let
+  # might become a problem with electron once this builds
+  # just a workaround anyway: https://github.com/svanderburg/node2nix/issues/106#issuecomment-418707444
+  nodejs = nodejs-6_x;
```
Comment (asymmetric) [inline pkgs/applications/networking/ssb/talenet/default.nix]: Could you check if this has been [solved](https://github.com/svanderburg/node2nix/issues/106#issuecomment-442952578)?

Comment (cryptix) [reply pkgs/applications/networking/ssb/talenet/default.nix]: I tried again with @svanderburg new version but still run into ENOTCACHED (ansi-styles).

My current WIP branch is on [this](https://github.com/talenet/talenet/commit/a9d6a5f1937fe9cbf1ef5f8146a10ae410d45f93) branch of tale:net.

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by cryptix** [comment]
Sadly no. This project isn’t under active development currently.

**Pull request closed without merging**